src/fft.rs

```diff
@@ -0,0 +1,56 @@
+use ark_bls12_381::Fr;
+use ark_ff::{Field, One};
+use ark_poly::{EvaluationDomain, Radix2EvaluationDomain};
+
+type Scalar = Fr;
+
+/// From dusk_bls12-381
+// group_gen, log_size_of_group, size
+pub fn domain(degree: usize) -> Radix2EvaluationDomain<Fr> {
+    Radix2EvaluationDomain::new(degree).unwrap()
+}
+
+/// From dusk_bls12-381
+pub fn fft(a: &mut [Scalar], omega: Scalar, log_n: u32) {
```

Comment:
btw, the Arkworks FFT implementation should be faster. If the domain API is inconvenient, we can expose a function like this to Arkworks